DSharpPlus.Commands/Processors/SlashCommands/InteractionNamingPolicies/KebabCaseNamingPolicy.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using DSharpPlus.Commands.Trees;
+
+namespace DSharpPlus.Commands.Processors.SlashCommands.NamingPolicies;
+
+/// <summary>
+/// Transforms parameter names into kebab-case.
+/// </summary>
+public sealed class KebabCaseNamingPolicy : IInteractionNamingPolicy
+{
+    /// <summary>
+    /// Transforms the command name into it's kebab-case equivalent.
+    /// </summary>
+    /// <inheritdoc />
+    public string GetCommandName(Command command)
+    {
+        string commandName = command.Attributes.FirstOrDefault(attribute =>
+            attribute is DisplayNameAttribute
+        )
+            is DisplayNameAttribute displayNameAttribute
+            ? displayNameAttribute.DisplayName
+            : command.Name;

```

Comment:
why the double type check

Reply:
Because `command.Attributes.OfType<DisplayNameAttribute>().FirstOrDefault()` is double the iterations